kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/FunctionDefs.java

```diff
@@ -0,0 +1,136 @@
+import org.drools.javaparser.ast.expr.CastExpr;
+import org.drools.javaparser.ast.expr.ClassExpr;
+import org.drools.javaparser.ast.expr.Expression;
+import org.drools.javaparser.ast.expr.MethodCallExpr;
+import org.drools.javaparser.ast.expr.NameExpr;
+import org.drools.javaparser.ast.expr.NullLiteralExpr;
+import org.drools.javaparser.ast.expr.StringLiteralExpr;
+import org.drools.javaparser.ast.type.Type;
+
+public class FunctionDefs {
+
+    private static final String ANONYMOUS = "<anonymous>";
+    private static final Pattern METHOD_PARSER = Pattern.compile("(.+)\\((.*)\\)");
+    private static final Pattern PARAMETER_PARSER = Pattern.compile("([^, ]+)");
+    private static final Type TYPE_BIG_DECIMAL = JavaParser.parseType(BigDecimal.class.getCanonicalName());
+
+    public static Expression asMethodCall(
+            String className,
+            String methodSignature,
+            List<String> params) {
+        // creating a simple algorithm to find the method in java
+        // without using any external libraries in this initial implementation
+        // might need to explicitly use a classloader here
+        String[] mp = parseMethod(methodSignature);
+        try {
+            String methodName = mp[0];
+            List<String> paramTypeNames = parseParams(mp[1]);
+            ArrayList<Expression> paramExprs = new ArrayList<>();
+            if (paramTypeNames.size() == params.size()) {
+                for (int i = 0; i < params.size(); i++) {
+                    String paramName = params.get(i);
+                    String paramTypeName = paramTypeNames.get(i);
+                    Type paramTypeCanonicalName =
+                            JavaParser.parseType(
+                                    getType(paramTypeName).getCanonicalName());
+
+                    Expression param =
+                        new CastExpr(paramTypeCanonicalName,
+                            new MethodCallExpr(
+                                null,
+                                "coerceTo",
+                                new NodeList<>(
+                                        new ClassExpr(paramTypeCanonicalName),
+                                        new MethodCallExpr(
+                                            new NameExpr("feelExprCtx"),
+                                            "getValue",
+                                            new NodeList<>(new StringLiteralExpr(paramName))
+                                        ))));
+
+                    paramExprs.add(param);
+                }
+
+                return new MethodCallExpr(
+                        new NameExpr(className),
+                        methodName,
+                        new NodeList<>(paramExprs));
+            } else {
+                return new NullLiteralExpr();
+            }
+        } catch (Exception e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    private static Class<?> getType(String typeName)
```

Comment:
this `getType` and the others below, please just convert existing as `public static`.

Reply:
(from original `FunctionDefNode`)